### ISSUE TITLE
Is there any intention of generating either a Universal Binary or Apple Silicon version of AutoPkgr?

### ISSUE THREAD

**Issue by fseesink** [lindegroup/autopkgr#708]
This is really more just a question/feature request.  Currently AutoPkgr is compiled as an Intel app.  I am just curious whether a Universal Binary version or Apple Silicon version of AutoPkgr is in the works.

Right now it's not an issue as we have Rosetta2.  But much like the transition from PowerPC to Intel eventually led to the release of macOS 10.6 "Snow Leopard"--when Apple pruned all the PowerPC code out of the OS and eventually removed Rosetta (which then allowed PowerPC binaries to run on Intel hardware)--we can expect Apple to do the same again at some point, stripping out the x64 code and removing Rosetta2.  At which point apps like AutoPkgr stop functioning altogether.

**Comment by shawnhonsberger** [comment]
Hi @fseesink unfortunately, this is not possible due to the fact that many of the libraries AutoPkgr depends on are not available as a Universal Binary. That being said, I think we are several years away from Rosetta 2 being removed.

**Comment by fseesink** [comment]
That's unfortunate.

And I do hope you're right.

But I wouldn't be too sure about that.  If you know your Apple history, we're a lot closer than you think.

The first Intel Mac was released in Jan 2006.  And among those was the Intel CoreDuo (note: not Core2Duo) 15" MacBook Pro.  I know, as I owned one.  And it came with Mac OS X 10.4.4 "Tiger".

macOS 10.6 "Snow Leopard" was released in Sep 2009, and it was the version where they had no real new features, but they had spent 1.5 years simply "tuning" the OS.  They stripped out all the PowerPC code from the OS, effectively removing the Universal Binary files in the OS, cutting required disk space for the OS in half.  And this was the last version with Rosetta support.

macOS 10.7 "Lion" was released in Jul 2011.  That was the first version where you no longer had Rosetta.  So from launch to dropping support for the PowerPC chip == 5.5 years.

Now the first M1 Macs were released in Nov 2020.  We're already in the 4th year of Apple Silicon.  And the rate of execution is much faster than with Intel.  We now have annual OS updates, and the M series are being released faster, not slower, than their Intel brethren.

Even if we assume Apple holds off the same amount of time, we are only ~2 years away.  And frankly, I don't see it lasting that long.  I can see them announcing that they're going to strip x64 code out of the next version of macOS, doing the equivalent of what they did with "Snow Leopard".  And then the clock is really ticking.

Apple has a known history here.  And twice, once they felt they were far enough along, they gutted Rosetta support.  Keep in mind this is the THIRD time they've done this.  First the transition from Motorola 68K to PowerPC.  (Ironically I had the very first PowerPC Mac, the PowerMac 6100, on my desk as well.  So I very vividly remember what that experience was like.)  Then PowerPC to Intel.  Now Intel to ARM64.  And each time Apple hit a point where they removed Rosetta.  I don't see the pattern changing here.

But again, I hope you're right.  I would love nothing better than to be wrong on this.
